packages/circuit-ui/components/NotificationModal/NotificationModal.tsx

```diff
@@ -110,68 +83,59 @@ export const NotificationModal: ModalComponent<NotificationModalProps> = ({
   preventClose = false,
   className,
   ...props
-}) => {
-  if (process.env.NODE_ENV !== 'production' && className) {
-    throw new CircuitError(
-      'NotificationModal',
-      'Custom styles are not supported by this component. If your use case requires custom styles, please open an issue at https://github.com/sumup-oss/circuit-ui.',
-    );
-  }
-
-  const reactModalProps = {
-    className: {
-      base: classes.base,
-      afterOpen: classes.open,
-      beforeClose: classes.closed,
-    },
-    overlayClassName: {
-      base: classes.overlay,
-      afterOpen: classes.open,
-      beforeClose: classes.closed,
-    },
-    onRequestClose: onClose,
-    closeTimeoutMS: TRANSITION_DURATION,
-    shouldCloseOnOverlayClick: !preventClose,
-    shouldCloseOnEsc: !preventClose,
+}: NotificationModalProps) => {
+  const headlineId = useId();
+  const initialFocusRef = useRef<HTMLButtonElement>(null);
+  const dialogProps = {
+    className: clsx(classes.centered, className, classes.base),
+    closeButtonLabel,
+    'aria-labelledby': headlineId,
+    preventClose,
+    onClose,
+    initialFocusRef,
     ...props,
   };
 
   function wrapOnClick(onClick?: ButtonProps['onClick']) {
     return (event: ClickEvent) => {
-      onClose?.(event);
+      onClose?.();
       onClick?.(event);
     };
   }
 
   return (
-    <ReactModal {...reactModalProps}>
-      {!preventClose && closeButtonLabel && (
-        <CloseButton onClick={onClose} className={classes.close}>
-          {closeButtonLabel}
-        </CloseButton>
+    <Dialog {...dialogProps}>
+      {() => (
+        <>
+          <NotificationImage image={image} />
+          <Headline
+            as="h2"
+            size="s"
+            id={headlineId}
+            className={classes.headline}
+          >
+            {headline}
+          </Headline>
+          {body && <Body>{body}</Body>}
+          {actions && (
+            <ButtonGroup
+              actions={{
+                primary: {
+                  ...actions.primary,
+                  onClick: wrapOnClick(actions.primary.onClick),
+                },
+                secondary: actions.secondary && {
+                  ...actions.secondary,
+                  // @ts-expect-error ref is a valid prop on a button
+                  ref: initialFocusRef,
```

Comment:
Why is the secondary action focused instead of the primary one?

Reply:
I am assuming that the secondary action is usually less destructive than primary (cancel, close...)

Reply:
Good point. I've seen both on macOS, though if we don't want to make this configurable, this is probably the safer option.

We should check the focus order of the ButtonGroup with Caleb or Steven at some point. The visual order differs between wide and narrow viewports, but the primary action always comes first in the focus order.